### PR TITLE
Add court type, fix ordering

### PR DESCRIPTION
**Pull request by jasonf7**

### CODE REVIEW

File: src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala [hunk 1]
```diff
@@ -95,33 +99,45 @@ object MatchMaker {
...
-          courtId = courtId
+          courtId = courtId,
+          courtType = MatchType.toType(courtSize)
         )
       }
+      .sortBy(_.courtId)
 
     println("***** MATCHES *****")
-    matches.foreach(println)
+    matches.foreach { lol =>
```
Comment (clemmy) [inline src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala:138]: lol

File: src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala [hunk 2]
```diff
@@ -95,33 +99,45 @@ object MatchMaker {
...
     }
 
     // 6. Map the chunks into matches
-    matches = chunks
-      .map { players => players.splitAt(players.size/2) }
+    val sortedCourtTypes = courtTypes
       .zipWithIndex
-      .map { case ((team1, team2), ind) =>
-        val courtId = ind + 1
+      .map { case (matchType, ind) =>
```
Comment (clemmy) [inline src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala:116]: nice

File: src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala [hunk 3]
```diff
@@ -95,33 +99,45 @@ object MatchMaker {
...
+        (ind + 1, MatchType.toSize(matchType))
+      }
+      .sortBy( - _._2)
```
Comment (clemmy) [inline src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala:119]: holy shit what is this

File: src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala [hunk 4]
```diff
@@ -95,33 +99,45 @@ object MatchMaker {
...
+
+    val playerTeams = chunks
+      .map { players => players.splitAt(players.size/2) }
+      .padTo(sortedCourtTypes.length, (Iterable(), Iterable()))
```
Comment (clemmy) [inline src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala:123]: What are the values in the next part whe (Itenrable(), Iterable()) is matched with (courtId, courtSize)?

File: src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala (hunk 3 above)
Comment (jasonf7) [reply src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala:119]: sort by match size descending

File: src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala (hunk 2 above)
Comment (jasonf7) [reply src/main/scala/com/shuttleql/services/game/matchmaking/MatchMaker.scala:116]: i'll assume that's **not** sarcastic LOL

**Review by clemmy — APPROVED** [review]

**Pull request merged**